File: src/commands/projects.ts
import { jira } from "../jira";
import { TablePrinter } from "../formatter";
import { CommandNode } from "../types";

export function projects(_: any, printer: TablePrinter): CommandNode {
  return {
    name: "projects",
    description: "List projects visible to the authenticated user.",
    usage: "jiri projects [options]",
    flags: [],
    run: async (_args, opts) => {
      printer.setDefaults(opts);
      const data = await jira.projects();
      const projects = data.values ?? [];
      const rows = [
        ["KEY", "NAME"],
        ...projects.map((p: any) => [p.key as string, p.name as string]),
      ];
      console.log(printer.render(rows));
    },
  };
}

File: src/commands/projects.rs
use crate::client::AtlassianClient;
use crate::formatter::Formatter;

/// Execute the projects command to list projects.
pub async fn run(client: &AtlassianClient, formatter: &Formatter) -> Result<(), String> {
    let projects = client.projects_all().await?;

    let mut rows = vec![vec!["KEY".to_string(), "NAME".to_string()]];
    for p in projects {
        rows.push(vec![
            p["key"].as_str().unwrap_or_default().to_string(),
            p["name"].as_str().unwrap_or_default().to_string(),
        ]);
    }

    println!("{}", formatter.render(rows));
    Ok(())
}
